### PR TITLE
SOCKS5 support to echo_client sample

### PR DESCRIPTION
**Pull request by rveerama1**
SOCKS5 support added to echo_client sample. Details are added in README file about how to verify this feature using echo-client with echo-server running on Linux host.

### CODE REVIEW

**Review by jukkar — APPROVED** [review]
LGTM

File: samples/net/sockets/echo_client/README.rst [hunk 1]
```diff
@@ -90,6 +90,36 @@ The certificate and private key used by the sample can be found in the sample's
...
+====================
+
+It is also possible to connect to the echo-server through a SOCKS5 proxy.
+To enable it, use ``-DOVERLAY_CONFIG=overlay-socks5.conf`` when running ``west
+build`` or  ``cmake``.
+
+By default, to make the testing easier, the proxy is expected to run on the
+same host as the echo-server in Linux host.
+
+To start a proxy server, ``ssh`` can be used.
```
Comment (pfalcon) [inline samples/net/sockets/echo_client/README.rst]: Can you reword this to make it clear that ssh has builtin SOCKS server support. E.g. "To start a proxy server, for example a builtin SOCKS server support in ``ssh`` can be used (`-D` option)."

File: samples/net/sockets/echo_client/README.rst [hunk 2]
```diff
@@ -90,6 +90,36 @@ The certificate and private key used by the sample can be found in the sample's
...
+Use the following command to run it on your host with the default port:
+
+.. code-block: console
+
+        $ ssh -N -D 0.0.0.0:1080 localhost
+
+.. code-block: console
+
+        $ ssh -N -D [::]:1080 localhost
```
Comment (pfalcon) [inline samples/net/sockets/echo_client/README.rst]: I'd suggest to explicitly say this command is for proxy on IPv6 address. And also, say explicitly of both commands need to be run at the same time (I for one would run only one for IPv4 for starters).

File: samples/net/sockets/echo_client/src/tcp.c [hunk 3]
```diff
@@ -91,6 +91,44 @@ static int start_tcp_proto(struct data *data, struct sockaddr *addr,
 		return -errno;
 	}
 
+#if defined(CONFIG_SOCKS)
+#define SOCKS5_PROXY_V6_ADDR CONFIG_NET_CONFIG_PEER_IPV6_ADDR
```
Comment (pfalcon) [inline samples/net/sockets/echo_client/src/tcp.c]: Can we please move these defines to the top of the file? (With a comment that they are used only for CONFIG_SOCKS).

**Review by pfalcon — COMMENTED** [review]
Looks good, some cleanup/clarify suggestions.

**Comment by rveerama1** [comment]
@pfalcon updated, please take a look.

**Review by pfalcon — APPROVED** [review]
Looks good, thanks!

File: samples/net/sockets/echo_client/src/tcp.c [hunk 4]
```diff
@@ -91,6 +102,41 @@ static int start_tcp_proto(struct data *data, struct sockaddr *addr,
 		return -errno;
 	}
 
+#if defined(CONFIG_SOCKS)
```
Comment (jukkar) [inline samples/net/sockets/echo_client/src/tcp.c]: Could we change this to `if (IS_ENABLED(CONFIG_SOCKS5) { `

Comment (rveerama1) [reply samples/net/sockets/echo_client/src/tcp.c]: Done.

File: samples/net/sockets/echo_client/README.rst [hunk 5]
```diff
@@ -90,6 +90,43 @@ The certificate and private key used by the sample can be found in the sample's
...
+For IPv6 proxy server:
+
+.. code-block: console
+
+        $ ssh -N -D [::]:1080 localhost
+
+Run both commands if you are testing IPv4 and IPv6.
+
+To connect to a proxy server that is not running under the same IP as the
+echo-server or uses a different port number, modify the following values:
```
Comment (dbkinder) [inline samples/net/sockets/echo_client/README.rst]: mention which file to make this modification: ``echo_client/src/tcp.c``

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: samples/net/sockets/echo_client/README.rst (hunk 5 above)
Comment (rveerama1) [reply samples/net/sockets/echo_client/README.rst]: Done, please take a look.

**Comment by jukkar** [comment]
I think we should get this to 2.0 so that there is an example how to enable proxy support when not using MQTT. Adding TSC label.

**Review by rlubos — APPROVED** [review]
LGTM

**Comment by pfalcon** [comment]
> I think we should get this to 2.0 so that there is an example... Adding TSC label.

I'd assume examples for existing features don't qualify as new features themselves, and don't need TSC approval? @ioannisg ?

**Comment by ioannisg** [comment]
@dbkinder could you revisit your review?

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, thanks!
